ui/cypress/e2e/queryBuilder.test.ts

```diff
@@ -0,0 +1,141 @@
+  const digits = []
+  for (let i = 0; i < 6; i++) {
+    digits[i] = Math.floor(Math.random() * Math.floor(10))
+  }
+  return Number(digits.join(''))
+}
+
+describe('The Query Builder', () => {
+  beforeEach(() => {
+    cy.flush()
+
+    cy.signin().then(({body}) => {
+      cy.wrap(body.org).as('org')
+    })
+
+    cy.writeData([
+      `mem,host=thrillbo-swaggins active=${generateRandomSixDigitNumber()}`,
+      `mem,host=thrillbo-swaggins cached=${generateRandomSixDigitNumber()}`,
+
+      `mem,host=thrillbo-swaggins active=${generateRandomSixDigitNumber()}`,
+      `mem,host=thrillbo-swaggins cached=${generateRandomSixDigitNumber()}`,
+    ])
+  })
+
+  describe('from the data explorer screen', () => {
+    it('creates a query, edits it to add another field, then views its results with pride and satisfaction', () => {
+      cy.get('@org').then((org: Organization) => {
+        cy.visit(`orgs/${org.id}/data-explorer`)
+      })
+
+      cy.contains('mem').click('right') // users sometimes click in random spots
+      cy.contains('active').click('bottomLeft')
+
+      cy.getByTestID('empty-graph--no-queries').should('exist')
+
+      cy.contains('Submit').click()
+
+      cy.get('.giraffe-plot').should('exist')
+
+      cy.contains('Save As').click()
+
+      // open the dashboard selector dropdown
+      cy.contains('Choose at least 1 dashboard').click()
+      cy.contains('Create a New Dashboard').trigger('mousedown')
+      cy.getByTestID('overlay--container').click() // close the blast door i mean dropdown
+
+      cy.get('[placeholder="Add dashboard name"]').type('Basic Ole Dashboard')
+      cy.get('[placeholder="Add optional cell name"]').type('Memory Usage')
+      cy.contains('Save as Dashboard Cell').click()
+
+      // wait for the notification since it's highly animated
+      // we close the notification since it contains the name of the dashboard and interfers with cy.contains
+      cy.wait(250)
+      cy.get('.notification-close').click()
+      cy.wait(250)
+
+      // force a click on the hidden dashboard nav item (cypress can't do the hover)
+      cy.getByTestID('nav-menu_dashboard').click({force: true})
+
+      cy.contains('Basic Ole Dashboard').click()
+      cy.getByTestID('cell-context--toggle').click()
+      cy.contains('Configure').click()
+
+      cy.get('.giraffe-plot').should('exist')
+      cy.contains('cached').click()
+      cy.contains('mean').click()
+      cy.contains('Submit').click()
+      cy.getByTestID('save-cell--button').click()
+
+      cy.getByTestID('nav-menu_dashboard').click({force: true})
+
+      cy.contains('Basic Ole Dashboard').click()
+      cy.getByTestID('cell-context--toggle').click()
+      cy.contains('Configure').click()
+
+      cy.get('.giraffe-plot').should('exist')
+      cy.getByTestID('cancel-cell-edit--button').click()
+      cy.contains('Basic Ole Dashboard').should('exist')
+    })
+  })
+
+  describe('from the Dashboard screen', () => {
+    beforeEach(() => {
```

Comment:
why isn't this logic in the beforeEach above? it appears like all of these tests would need to follow a formula of
- flush
- signin
- goto page
- wait for availability
- [ application specific testing ]

Reply:
@drdelambre 

>why isn't this logic in the beforeEach above?

because one set of tests is going to `/dashboards` and one set of tests is going to `/data-explorer`. the other `describe` for `data-explorer` doesn't have a `beforeEach` (and this test probably doesn't need a `beforeEach` since there's only one test) and instead just [`visit`s `orgs/${org.id}/data-explorer`](https://github.com/influxdata/influxdb/pull/16134/files#diff-c432c510f4c140978dbe371ff3e9142dR31)

so the high level flow (after i remove the `beforeEach` here):

```
- describe the data-exporer test
  - flush
  - sign in
  - write a couple of metrics
  - visit `orgs/${org.id}/data-explorer`
  - assertions
- describe the dashboard test
  - flush
  - sign in
  - write a couple of metrics
  - visit `orgs/${org.id}/dashboard`
  - assertions
```

that's reasonable, right?

Reply:
totes! wasn't a critique, genuine curiosity. it feels like a pattern is forming among the tests we've written and i'm looking for data points to break my mental modal.